public/components/common/restart-modal/restart-modal.tsx

```diff
@@ -0,0 +1,205 @@
+      if (countDown === 0 && setState === setCountdown) {
+        clearInterval(interval);
+        RestartHandler.goToHealthcheck();
+      } else if (countDown === 0) {
+        clearInterval(interval);
+      }
+
+      countDown--;
+    }, 1000 /* 1 second */);
+  };
+  
+  // TODO review if importing these functions in wz-restart work.
+  const dispatch = useDispatch();
+  const updateRedux = {
+    updateRestartAttempt: (restartAttempt) => dispatch(updateRestartAttempt(restartAttempt)),
+    updateSyncCheckAttempt: (syncCheckAttempt) =>
+      dispatch(updateSyncCheckAttempt(syncCheckAttempt)),
+    updateUnsynchronizedNodes: (unsynchronizedNodes) =>
+      dispatch(updateUnsynchronizedNodes(unsynchronizedNodes)),
+    updateRestartStatus: (restartStatus) => dispatch(updateRestartStatus(restartStatus)),
+  };
+
+  const forceRestart = async () => {
+    await RestartHandler.restartWazuh(updateRedux);
+  };
+
+  // Build the modal depending on the restart state.
+  let emptyPromptProps: Partial<EuiEmptyPromptProps>;
+  switch (restartStatus) {
+    default:
+    case RestartHandler.RESTART_STATES.RESTARTING:
+      emptyPromptProps = {
+        title: (
+          <>
+            <img src={logotypeURL} className="wz-modal-restart-logo" alt=""></img>
+            <h2 className="wz-modal-restart-title">Restarting Wazuh</h2>
+          </>
+        ),
+        body: (
+          <>
+            <EuiProgress value={attempts} max={maxAttempts} size="s" />
+          </>
+        ),
+      };
+      break;
+
+    case RestartHandler.RESTART_STATES.RESTART_ERROR:
+      emptyPromptProps = {
+        iconType: 'alert',
+        iconColor: 'danger',
+        title: <h2 className="wz-modal-restart-title">Unable to connect to Wazuh.</h2>,
+        body: <p>There was an error restarting Wazuh. The Healthcheck will run automatically.</p>,
+        actions: (
+          <EuiButton color="primary" fill href="#/health-check">
+            Go to Healthcheck ({timeToHC} s)
+          </EuiButton>
+        ),
+      };
+      break;
+
+    case RestartHandler.RESTART_STATES.SYNC_ERROR:
+      emptyPromptProps = {
+        iconType: 'alert',
+        iconColor: 'danger',
+        title: <h2 className="wz-modal-restart-title">Synchronization failed</h2>,
+        body: (
+          <p>
+            The nodes {unsyncedNodes.join(', ')} did not synchronize. Restarting Wazuh might set the
+            cluster into an inconsistent state. Close and try again later.
+          </p>
+        ),
+        actions: (
+          <EuiFlexGroup justifyContent="flexEnd">
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty color='danger' flush="both" onClick={forceRestart}>
+                Force restart
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}> 
+              <EuiButton color="primary" fill onClick={() => dispatch(updateRestartStatus(RestartHandler.RESTART_STATES.RESTARTED))}>
```

Comment:
Move this code to a function named `abort` that does all the cleansing (sets everything to the proper state).
```suggestion
              <EuiButton color="primary" fill onClick={abort}>
```